portiaui/app/services/annotation-structure.js

```diff
@@ -146,14 +147,39 @@ const ElementStructure = Ember.Object.extend({
                         selectorMatcher.unRegister(selector, setElements);
                     }
                     selector = annotation.get('selectorGenerator.selector');
+                    const repeatedSelector = annotation.get('containerGenerator.selector');
                     if (!annotation.get('isDeleted')) {
                         annotation.setProperties({
                             selector,
                             xpath: annotation.get('selectorGenerator.xpath')
                         });
                     }
+                    annotation.set('repeated', false);
                     if (selector) {
                         selectorMatcher.register(selector, setElements);
+                        const elements = selectorMatcher.query(selector);
+                        if (repeatedSelector && elements && elements.length > 1) {
+                            const repeatedContainers = annotation.get('containerGenerator.parent.repeatedContainers');
+                            if (repeatedContainers.length > 1) {
```

Comment:
this logic belongs in the selector generator. and we should just read repeated from it here